pkg/systemd/overview-cards/lastLogin.jsx

```diff
@@ -55,28 +56,25 @@ const LastLogin = () => {
     const [messages, setLoginMessages] = useState(null);
     const [name, setName] = useState(null);
 
-    useEffect(() => {
-        if (messages === null) {
-            const bridge = cockpit.dbus(null, { bus: "internal" });
-            bridge.call("/LoginMessages", "cockpit.LoginMessages", "Get", [])
-                    .then(reply => {
-                        const obj = JSON.parse(reply[0]);
-                        if (obj.version == 1) {
-                            setLoginMessages(obj);
-                        } else {
-                        // empty reply is okay -- older bridges just don't send that information
-                            if (obj.version !== undefined)
-                                console.error("unknown login-messages:", reply[0]);
-                        }
-                    })
-                    .catch(error => {
-                        console.error("failed to fetch login messages:", error);
-                    });
-        }
-        if (name === null) {
-            cockpit.user().then(user => setName(user.name));
+    useInit(() => (async () => {
```

Comment:
I think you can just fo `useInit(async () => { ... })`, no?

Reply:
This [doesn't work with useEffect()](https://devtrium.com/posts/async-functions-useeffect). `useObject()` also uses the return value of the `create` function (i.e. that arrow), but it mustn't interpret the returned promise as a destroy function. After reading the code I wasn't sure if it's safe, but as you wrote it, you tell me?

Reply:
There's precedent indeed:
```
❱❱❱ git grep useInit.*async
pkg/apps/application-list.jsx:    useInit(async () => {
pkg/users/users.js:    useInit(async () => {
```

Reply:
Updated!

Reply:
Unlike `useEffect`, `useInit` and `useObject` deliver the return value of the function back to the render function.

Indeed, the function runs immediately as a subroutine call of the render function, and not sometime after rendering is done.